spec/support/formatters/user_flow_formatter.rb

```diff
@@ -0,0 +1,65 @@
+# Capybara configuration
+Capybara.save_path = Rails.root.join('public', 'user_flows')
+Capybara::Screenshot.prune_strategy = :keep_last_run
+
+# Capybara's `save_page` does not return anything. This is a hack to
+# capture the path of the screenshot so that it can be embedded in
+# the HTML output of this formatter
+Capybara::Screenshot.after_save_html do |path|
+  filename = path.split('/').last
+  uri = Capybara.save_path.to_s.split('public').last + '/' + filename
+  # href = "<a href=\"#{uri}\">screenshot</a>"
+  # $cur_link = href
+  $cur_link = uri
+end
+
+
+class UserFlowFormatter < RSpec::Core::Formatters::DocumentationFormatter
+  # This registers the notifications this formatter supports, and tells
+  # us that this was written against the RSpec 3.x formatter API.
+  RSpec::Core::Formatters.register self, :initialize,
+                                         :example_passed,
+                                         :example_group_finished,
+                                         :example_group_started,
+                                         :stop
+
+
+  def initialize(output)
+    @html = '<html><body><h1>login.gov user flows</h1><br>'
+    @user_flows_html_file = Capybara.save_path.join('index.html').to_s
+    super
+  end
+
+  def example_passed(notification)
+    indent = ' ' * notification.example.metadata[:scoped_id].split(':').size
+    @html << "<br />"
+    @html << "#{indent}<a href=\"#{$cur_link}\">#{notification.example.description}</a>"
+
+    super
+  end
+
+  def example_group_started(notification)
+    indent = super
+    
+    @html << '<pre>'
+    @html << ' ' * indent
+    @html << notification.group.description
+    
+  end
+
+  def example_group_finished(notification)
+    @html << '</pre>'
+    super
+  end
+
+  def stop(notification)
+    puts "Complete!\n"
+    @html += '</body></html>'
```

Comment:
What do you think about making this with an ERB instead? Building up a list of examples and then rendering it all at the end?